### PR TITLE
python310Packages.channels: 3.0.5 -> 4.0.0, python310Packages.daphne: 3.0.2 -> 4.0.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
https://github.com/django/channels_redis/blob/main/CHANGELOG.txt
https://github.com/django/channels/blob/main/CHANGELOG.txt
https://github.com/django/daphne/blob/main/CHANGELOG.txt

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 197042` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>paperless</li>
    <li>paperless-ng</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>baserow</li>
    <li>paperless-ngx</li>
    <li>python310Packages.channels</li>
    <li>python310Packages.channels-redis</li>
    <li>python310Packages.daphne</li>
    <li>python39Packages.channels</li>
    <li>python39Packages.channels-redis</li>
    <li>python39Packages.daphne</li>
  </ul>
</details>

File: pkgs/applications/office/paperless-ngx/default.nix [hunk 1]
```diff
@@ -215,14 +217,21 @@ python.pkgs.pythonPackages.buildPythonApplication rec {
       --replace "--cov --cov-report=html" ""
   '';
 
+  disabledTests = [
+    # Tests require a running Redis instance
+    "test_consume_barcode_file"
+    "test_consume_barcode_tiff_file"
+    "test_consume_barcode_supported_no_extension_file"
+  ];
+
```
Comment (gador) [inline pkgs/applications/office/paperless-ngx/default.nix:226]: why this change? Before this PR, `paperless-ngx` tests did not need a running Redis instance and all tests pass before..

**Comment by gador** [comment]
Ok. Figured out that there is an error in the new versions and this is why the tests fail (as they should). Could you have a look at https://github.com/gador/nixpkgs/tree/pr-197042 and incorporate it in this PR?

**Comment by fabaff** [comment]
Overriding is for sure the better day to deal with the issue.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 197042` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>baserow</li>
    <li>paperless-ngx</li>
    <li>python310Packages.channels</li>
    <li>python310Packages.channels-redis</li>
    <li>python310Packages.daphne</li>
    <li>python39Packages.channels</li>
    <li>python39Packages.channels-redis</li>
    <li>python39Packages.daphne</li>
  </ul>
</details>
